src/Symfony/Component/HttpKernel/HttpCache/ResponseCacheStrategy.php

```diff
@@ -82,7 +82,8 @@ public function add(Response $response)
         }
 
         $this->storeRelativeAgeDirective('max-age', $response->headers->getCacheControlDirective('max-age'), $age);
-        $this->storeRelativeAgeDirective('s-maxage', $response->headers->getCacheControlDirective('s-maxage') ?: $response->headers->getCacheControlDirective('max-age'), $age);
+        $sharedMaxAge = $response->headers->hasCacheControlDirective('s-maxage') ? $response->headers->getCacheControlDirective('s-maxage') : $response->headers->getCacheControlDirective('max-age');
```

Comment:
What if the header exists but has no value? Either null or an empty string. Is that a case we should consider?

Reply:
Does #41663 resolve that question? Or am I missing some edge case?

Reply:
I could add a `(int)` cast here to make sure it's an int value, and `0` would be the most conservative setting.

More generally, I cannot tell in which places things might break if you set an invalid header that way. `\Symfony\Component\HttpFoundation\Response::setSharedMaxAge()` would not allow you to do so, so it would take you some lower-level methods (a generic `setHeader()` call) to run into this.

What do you think @Nyholm ?

Reply:
i like the `int` cast to be on the safe side. empty values should not exist according to the HTTP spec, but treating them as `0` will not accidentally cache anything.

Reply:
After taking a closer look, it's not that easy in that spot because `NULL` is a valid value too (if the header is not set).

So, could we keep this out of this PR and maybe have a dedicated case for a review if it is possible to pass invalid values for these headers and to discuss how we could deal with (or prevent) that?

Reply:
hm, but as you first do the hasCacheControlDirective i would assume that if the header is not set, that check would return `false` and we read max-age. if you put the cast to int in the fragment of s-maxage that should work fine.

Reply:
See 8319c6954254a972204198c47357ce659ae57277 – does that do it?